onnxruntime/core/providers/cpu/math/det.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cpu/math/det.h"
+#include "core/util/math_cpuonly.h"
+
+using namespace onnxruntime::common;
+
+namespace onnxruntime {
+
+ONNX_CPU_OPERATOR_KERNEL(
+    Det,
+    11,
+    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    Det<float>);
+
+template <typename T>
+Status Det<T>::Compute(OpKernelContext* context) const {
+  const auto* X = context->Input<Tensor>(0);
+  ORT_ENFORCE(X != nullptr);
+
+  const auto& X_shape = X->Shape();
+  int X_num_dims = static_cast<int>(X_shape.NumDimensions());
+
+  // input validation
+  if (X_num_dims < 2) {  // this is getting capture by shape inference code as well
```

Comment:
Should we also check for X_num_dims > 3, given that spec does not allow more than >3D tensor as input?

Reply:
good point. sure

Reply:
no in fact the spec allows >3D tensors. For rank > 3, axes [0, 1, ..., rank - 3] are all treated as batch axes.

Reply:
fixed